### PR TITLE
add GPU pipeline support based on xgboost 1.3 release

### PR DESCRIPTION
**Pull request by wbo4958**
1. enable gpu train/transform pipeline
2. support incremental dmatrix building
3. add xgboost pyspark supporting
4. add RMM supporting

this repo depends on NVIDIA/spark-rapids 0.3.0 release
and rapidsai/cudf 0.17 release

### CODE REVIEW

File: Jenkinsfile.automerge [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/usr/local/env groovy
```
Comment (firestarman) [inline Jenkinsfile.automerge]: Better to remove this Jenkinsfile.

**Review by firestarman — COMMENTED** [review]
Overall it LGTM except the comment as below.

**Comment by NvTimLiu** [comment]
+1

File: tests/ci_build/jenkins_tools_spark.Groovy [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/groovy
```
Comment (firestarman) [inline tests/ci_build/jenkins_tools_spark.Groovy]: If this is created by us, better to remove it

File: Jenkinsfile.automerge (hunk 1 above)
Comment (wbo4958) [reply Jenkinsfile.automerge]: Good findings

File: tests/ci_build/jenkins_tools_spark.Groovy (hunk 2 above)
Comment (wbo4958) [reply tests/ci_build/jenkins_tools_spark.Groovy]: Good findings, Done

File: Jenkinsfile.automerge (hunk 1 above)
Comment (wbo4958) [reply Jenkinsfile.automerge]: Done

**Review by firestarman — APPROVED** [review]

**Pull request merged**